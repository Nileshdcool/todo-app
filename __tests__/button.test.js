import React from 'react';
import renderer from 'react-test-renderer';
import FlatButton from '../components/UI/AuthFlatButton';

describe('FlatButton Component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<FlatButton onPress={() => {}}>Hello</FlatButton>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls the onPress function when pressed', () => {
    const onPressMock = jest.fn();
    const component = renderer.create(<FlatButton onPress={onPressMock}>Press Me</FlatButton>);
    const pressable = component.root.findByType(FlatButton);

    // Simulate a button press by calling the onPress prop
    pressable.props.onPress();

    expect(onPressMock).toHaveBeenCalled();
  });
});
