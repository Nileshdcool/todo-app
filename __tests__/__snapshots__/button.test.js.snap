// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlatButton Component renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "paddingHorizontal": 12,
        "paddingVertical": 6,
      },
      false,
    ]
  }
>
  <View>
    <Text
      style={
        {
          "color": "#f9beda",
          "textAlign": "center",
        }
      }
    >
      Hello
    </Text>
  </View>
</View>
`;
